### ISSUE TITLE
Create posts online for working groups

### ISSUE THREAD

**Issue by starbelly** [erlef/website#200]
Working groups should be able to login to the site and easily write blog posts for their group. This would be achieved using live view for real time editing and preview,  optimally would have fallback to non-live view components for off-line editing and preview. 

- Anyone in a working group should be able to create a post for their working group.
- Peer or admin review is to be required before a post may be published.
- If a post is edited after publication, re-publishing should also require review. 

## Proposed flow 

 - Member logins to site
 - Member goes to a section on the site (i.e., working group ) or selects from a menu to add a new post
 - Member interacts with something similar to https://markdown.dichev.io/
 - Input must be sanitized using something like https://github.com/rrrene/html_sanitize_ex
 - When the member is done they will click save, content will go into a quarantine/queue table 
 - Upon clicking save the marketing group (and possibly other designated administrators) will be notified of a new a post in queue for review. 
- Review will consist of looking for egregious code of conduct (if you will and tbd) content violations
    - All reviews must be non-bias as possible, thus a would-be reviewer can not approve their own content, nor their own organizations content. 
- Once the content has at least N approvals (tbd) it will be saved to a posts table. 
- The post may be edited after publishing but must be subjected to the same review cycle. 

This is issue is blocked by #199

**Comment by tsloughter** [comment]
If this is one of the ways the website ppl saw working groups contributing to the site then its clear why they haven't and is more reason why I dislike developing a whole website for this instead of using a static site.

I'm not saying WGs would but if it was just sending a markdown PR to add a blog post it would be something that could be done today. Now it isn't.

**Comment by starbelly** [comment]
@tsloughter What we have observed is that some people prefer not to interact with git etc. This ranges from engineers to non-engineers, more understandable with the latter. It's possible we can provide a way to do both, that may or may not be a good idea, so I don't think it should be taken off the table.

**Comment by peerst** [comment]
I think its less the "interact with git" but install Elixir and some Elixir code and run it before committing

**Comment by tsloughter** [comment]
Should just be markdown that can either be sent as a PR or an email. its not like we want to give everyone permission to publish anyway. So maybe if we had wordpress we could give every working group perms to create articles but not make them published and then owners of the site can publish them. But if not going with email I think markdown through git or email makes the most sense.

**Comment by paulo-ferraz-oliveira** [comment]
I agree with @tsloughter's idea. Markdown's OK. Even if it was an Issue in markdown that one of us later converted to a pull request, it's still better than having less than that.

**Comment by starbelly** [comment]
Yeah, so this issue is stale, sort of, let me see if I can bring it current.

 - I've already done the work to do blog support "online", but we stopped and asked ourselves, are we going out of our wheelhouse now? And we think the answer might be yes. It doesn't mean that this isn't in our future, but there's more important things to resolve right now. There wasn't a huge amount of effort put into the POC on my end because all the components sans the schema and some logic were there per other features on the site right now (i.e., the markdown editor for wg charters).
 - The initial thinking was yeah, just do markdown, commit to the repo, call it a day. This works great if you're a technical person, but it doesn't work at all for non-technical people. They literally have to install erlang, elixir, get the repo, install assets, etc. etc. 
 - Per the above we now think we should use a tried and true blogging platform. We are going to evaluate forestry and ghost. Ghost is looking more promising because it has an api vs having to hack an api in with a static site produced by something like forestry. But, more suggestions are great. 
 
 Here are the requirements as I see it: 
 
  - All the things you expect out of a basic blogging platform (wysiwyg editor, media library, etc.).
  - Easily be able to unify the look with the main site
  - Permissions (i.e., if one team creates a post, have the ability to let anyone in that team edit the post, etc.)

**Comment by starbelly** [comment]
I'm going to close this for now in favor of the following discussion https://github.com/erlef/infra-wg/discussions/47
